src/hooks/useENS.ts

```diff
@@ -0,0 +1,24 @@
+import { useEffect, useState } from "react"
+import { ethers } from "ethers"
+
+type ReturnType = { ensName: string | null }
+
+export const useENS = (address: string | null | undefined): ReturnType => {
+  const [ensName, setENSName] = useState<string | null>(null)
+
+  useEffect(() => {
+    async function resolveENS() {
+      if (address && ethers.utils.isAddress(address)) {
+        const provider = new ethers.providers.JsonRpcProvider(
```

Comment:
Prefer to use provider library from `const { library } = useActiveWeb3React()` instead of using Env variable